server/compose/service/attachment.go

```diff
@@ -685,6 +692,25 @@ func (svc attachment) processImage(original io.ReadSeeker, att *types.Attachment
 	return svc.objects.Save(att.PreviewUrl, buf)
 }
 
+// processIcon processes an icon attachment and prepares preview by using original meta for the icon.
+// solution for broken preview on UI for now.
+func (svc attachment) processIcon(original io.ReadSeeker, att *types.Attachment) (err error) {
+	if _, err = original.Seek(0, 0); err != nil {
+		return
+	}
+
+	if att.Meta.Preview == nil {
+		att.Meta.Preview = &att.Meta.Original
+		att.Meta.Preview.Image = &types.AttachmentImageMeta{
+			Width:    attachmentPreviewMaxWidth,
+			Height:   attachmentPreviewMaxHeight,
```

Comment:
This sounds unsafe to me -- shouldn't we get the dimensions from the icon itself?
If there are issues with that, we could also display icons like we do other non-image files (the little font awesome icon)